### PR TITLE
feat(sso): Add Google Auth Support

### PR DESCRIPTION
**Pull request by jdenquin**
## Context

- We want to support Login with Google Auth for existing users

## Description

- Add resolvers and mutations on graphQL for Google Auth Support
- Add services for google auth

### CODE REVIEW

File: app/graphql/resolvers/auth/google/auth_url_resolver.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+module Resolvers
+  module Auth
+    module Google
+      class AuthUrlResolver < Resolvers::BaseResolver
+        graphql_name 'GooGleAuthUrl'
```
Comment (ansmonjol) [inline app/graphql/resolvers/auth/google/auth_url_resolver.rb:7]: Typo here

**Review by ansmonjol — APPROVED** [review]

**Review by vincent-pochet — APPROVED** [review]

File: app/services/auth/google_service.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      @client_id = Google::Auth::ClientId.new(ENV['GOOGLE_AUTH_CLIENT_ID'], ENV['GOOGLE_AUTH_CLIENT_SECRET'])
+
+      super
+    end
+
+    def authorize_url(request)
+      authorizer = Google::Auth::WebUserAuthorizer.new(
+        client_id,
+        BASE_SCOPE,
+        nil,
```
Comment (rsempe) [inline app/services/auth/google_service.rb:17]: maybe a comment to name this parameter? 😅

Comment (jdenquin) [reply app/services/auth/google_service.rb:17]: its a nil parameter, no one knows about it 😂

File: spec/services/auth/google_service_spec.rb [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+      result = service.authorize_url(request)
+
+      aggregate_failures do
+        expect(result).to be_success
+        expect(result.url).to include('https://accounts.google.com/o/oauth2/auth')
+      end
+    end
+  end
+
+  describe '.login' do
```
Comment (rsempe) [inline spec/services/auth/google_service_spec.rb:25]:
```suggestion
  describe '#login' do
```

**Review by rsempe — APPROVED** [review]

**Pull request merged**